src/pages/profile/components/profile-product/product-management/index.tsx

```diff
@@ -0,0 +1,178 @@
+	} = data || { products: [], totalCount: 0, totalPages: 0 };
+
+	useEffect(() => {
+		refetch();
+	}, [searchQuery, filter, order, page, refetch]);
+
+	useEffect(() => {
+		updateTotalPages(fetchedTotalPages || pagination.DEFAULT_PAGE);
+	}, [fetchedTotalPages, updateTotalPages]);
+
+	const handleSearch = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>) => {
+			setSearchQuery(event.target.value);
+		},
+		[],
+	);
+
+	const handleChangeSort = useCallback(() => {
+		setOrder((prevOrder) => (prevOrder === sortOrder.ASC ? "DESC" : "ASC"));
+		refetch();
+	}, [refetch]);
+
+	const handleFilterChange = useCallback(
+		(event: SelectChangeEvent<unknown>) => {
+			setFilter(event.target.value as string);
+			refetch();
+		},
+		[refetch],
+	);
+
+	const getOrderDisplayText = (order: "ASC" | "DESC"): string => {
+		return order === sortOrder.DESC
+			? t("VendorProfilePage.ascending")
+			: t("VendorProfilePage.descending");
+	};
+
+	return (
+		<StyledContainer>
+			<Typography
+				sx={{
+					fontFamily: theme.typography.playfairDisplayBold,
+					fontSize: theme.fontSizes.large,
+				}}
+			>
+				{t("AdminPage.products")}
+				<StyledProductsCount>{`: ${totalCount}`}</StyledProductsCount>
+			</Typography>
+			<StyledWrapperContainer>
+				<StyledWrapperHeader>
+					<Box sx={{ alignItems: "center", display: "flex" }}>
+						<BoldDivider />
+						<StyledHeaderTypography>
+							{t("AdminPage.productList")}
+						</StyledHeaderTypography>
+					</Box>
+					<Box sx={{ display: "flex", gap: "10px" }}>
+						<StyledSortButton onClick={handleChangeSort}>
+							<SortIcon
+								sx={{
+									transform:
+										order === sortOrder.ASC ? "rotate(180deg)" : undefined,
+								}}
+							/>
+							{getOrderDisplayText(order)}
+						</StyledSortButton>
+						<StyledSortSelect
+							displayEmpty
+							onChange={handleFilterChange}
+							startAdornment={
+								<InputAdornment position="start">
+									<FilterAltIcon sx={{ cursor: "pointer", width: "40px" }} />
+								</InputAdornment>
+							}
+							value={filter}
+						>
+							<MenuItem value="name">{t("VendorProfilePage.name")}</MenuItem>
+							<MenuItem value="category">
+								{t("VendorProfilePage.category")}
+							</MenuItem>
+							<MenuItem value="status">
+								{t("VendorProfilePage.status")}
+							</MenuItem>
+							<MenuItem value="price">{t("VendorProfilePage.price")}</MenuItem>
+						</StyledSortSelect>
+					</Box>
+				</StyledWrapperHeader>
+				<StyledSearchBox>
+					<StyledPaper component="form">
+						<IconButton aria-label="search" sx={{ p: "10px" }} type="submit">
+							<SearchIcon />
+						</IconButton>
+						<InputBase
+							onChange={handleSearch}
+							placeholder={t("AdminProductManagement.search")}
+							sx={{ width: "100%" }}
+							value={searchQuery}
+						/>
+					</StyledPaper>
+				</StyledSearchBox>
+				<ProductsTable products={fetchedProducts} />
+				{isLoading && <Loader />}
+				{isError && (
```

Comment:
I think it make sense to make a wrapper as in many components you have the same logic

Reply:
May i add the custom component for handle Loading and Error by next PR sb-67 ?
It will be necessary to go through the entire project and fix it everywhere...